### PR TITLE
python3Packages.pyopengl: fix substituteInPlace

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

Pyopengl is broken on `platform/glx`, which is what x86_64-linux seems to default to.
The substitutions were broken in 726628629f7ef3bf9b392e9f89b5a60b22f9836f during `python-updates`.
This would've been caught with #260776. _(please review it xoxo)_

I also took the liberty to fix some of the erroneous and missing patching I found. Now `pkgs.libglvnd` and `pkgs.gle` have become a part of the runtime closure.
For the interested, [here](https://gist.github.com/pbsds/116ba90c8362659188b2d42d7edb46ae) is a diff in `lib/python3.12/site-packages/OpenGL/platform` between v3.1.6 from before 726628629f7ef3bf9b392e9f89b5a60b22f9836f and v3.1.7 with these fixes applied.

Unbreaks https://hydra.nixos.org/build/246287273 and likely many others

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by a-n-n-a-l-e-e** [comment]
> Pyopengl is broken on `platform/glx`, which is what x86_64-linux seems to default to. The substitutions were broken in [7266286](https://github.com/NixOS/nixpkgs/commit/726628629f7ef3bf9b392e9f89b5a60b22f9836f) during `python-updates`. This would've been caught with #260776. _(please review it xoxo)_

i wonder if a few of the tests under `tests` can be enabled.

**Review by ghost — APPROVED** [review]
Failures seem unrelated -- mix of gcc13 errors, python missing dependencies, etc. 

Result of `nixpkgs-review pr 281017` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.stytra</li>
    <li>python311Packages.stytra.dist</li>
  </ul>
</details>
<details>
  <summary>15 packages failed to build:</summary>
  <ul>
    <li>gnss-sdr</li>
    <li>gnuradio3_9</li>
    <li>gnuradio3_9Packages.grnet</li>
    <li>gnuradio3_9Packages.osmosdr</li>
    <li>gnuradio3_9Packages.osmosdr.dev</li>
    <li>ib-controller</li>
    <li>inkcut</li>
    <li>inkcut.dist</li>
    <li>inkscape-extensions.inkcut</li>
    <li>linien-gui</li>
    <li>linien-gui.dist</li>
    <li>napari (python311Packages.napari)</li>
    <li>napari.dist (python311Packages.napari.dist)</li>
    <li>python311Packages.enamlx</li>
    <li>python311Packages.enamlx.dist</li>
  </ul>
</details>
<details>
  <summary>44 packages built:</summary>
  <ul>
    <li>friture</li>
    <li>friture.dist</li>
    <li>gnuradio</li>
    <li>gnuradioMinimal (gnuradio3_9Minimal)</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mymcplus</li>
    <li>mymcplus.dist</li>
    <li>opcua-client-gui</li>
    <li>opcua-client-gui.dist</li>
    <li>plots</li>
    <li>plots.dist</li>
    <li>printrun</li>
    <li>printrun.dist</li>
    <li>python311Packages.binwalk-full</li>
    <li>python311Packages.binwalk-full.dist</li>
    <li>python311Packages.openusd</li>
    <li>python311Packages.openusd.doc</li>
    <li>python311Packages.orange-widget-base</li>
    <li>python311Packages.orange-widget-base.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.pyopengl</li>
    <li>python311Packages.pyopengl.dist</li>
    <li>python311Packages.pyqtgraph</li>
    <li>python311Packages.pyqtgraph.dist</li>
    <li>python311Packages.pyrender</li>
    <li>python311Packages.pyrender.dist</li>
    <li>spyder (python311Packages.spyder)</li>
    <li>spyder.dist (python311Packages.spyder.dist)</li>
    <li>python312Packages.pyopengl</li>
    <li>python312Packages.pyopengl.dist</li>
    <li>run-scaled</li>
    <li>steamos-devkit</li>
    <li>steamos-devkit.dist</li>
    <li>tribler</li>
    <li>video2midi</li>
    <li>xpra</li>
    <li>xpra.dist</li>
  </ul>
</details>

**Comment by 7c6f434c** [comment]
(homepage change verified and OK)

**Comment by pbsds** [comment]
Do you want to test that shiny new commit bit @a-n-n-a-l-e-e?

**Comment by a-n-n-a-l-e-e** [comment]
> Do you want to test that shiny new commit bit @a-n-n-a-l-e-e?

note to self: check that builds actually kicked off before merging...

**Comment by pbsds** [comment]
FYI https://github.com/NixOS/ofborg/issues/577

**Comment by Sigmanificient** [comment]
This package does not work properly:
- #403733

**Pull request merged**
